### ISSUE TITLE
Trouble with RMarkdown files

### ISSUE THREAD

**Issue by JakobGM** [jalvesaq/Nvim-R#393]
# Trouble with RMarkdown files

## Summary

Hi!

Nvim-R works great for `.R`-files, but I am having some difficulties with `.Rmd` files. Most commands do not work, at it seems to be a problem with vimscript commands that need to be defined for rmarkdown files.

## Examples



When I trigger `\cc`, I get the following error message:

`E884: Function name cannot contain a colon: b:SendChunkToR("silent", "stay")`

While `\ss` results in:

```
Error detected while processing function SendSelectionToR:                                                                                                                
line    5:                                                                                                                       
E117: Unknown function: b:IsInRCode
```

And `\bb` results in

```
Error detected while processing function SendMBlockToR:                                                                                                                                                            
line    1:                                                                                                                                                                                                         
E117: Unknown function: b:IsInRCode                                                                                                                                                                                
E15: Invalid expression: &filetype != "r" && b:IsInRCode(1) == 0    
```

## Hypothesis

Seems like Nvim-R can't detect when the cursor is placed in an `r` chunk. That is why much of the functionality does not work. `\l`, which "location naive" *does work*. This issue also affects `ncm-R`, no autocompletion is available in Rmarkdown files.

## Extra information

* NVIM v0.3.4
* R version 3.5.2
* I had to manually build the `nvimcom` package.
* Linux v.5.0

Result of `:checkhealt`

```

health#vimtex#check
========================================================================
## vimtex
  - OK: Vim version should have full support!
  - OK: General viewer should work properly!
  - WARNING: Compiler callbacks will not work!
    - ADVICE:
      - `neovim-remote` / `nvr` is required for callbacks to work with neovim
      - Please also set |g:vimtex_compiler_progname| = 'nvr'
      - For more info, see :help |vimtex-faq-neovim|

health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=\177
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $COLORTERM='truecolor'

health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: xclip

## Python 2 provider (optional)
  - INFO: pyenv: Path: /opt/pyenv/libexec/pyenv
  - INFO: pyenv: Root: /opt/pyenv
  - WARNING: No Python executable found that can `import neovim`. Using the first available executable for diagnostics.
  - ERROR: Python provider error:
    - ADVICE:
      - provider/pythonx: Could not load Python 2:
          /usr/bin/python2 does not have the "neovim" module. :help |provider-python|
          /usr/bin/python2.7 does not have the "neovim" module. :help |provider-python|
          python2.6 not found in search path or not executable.
          /usr/bin/python is Python 3.7 and cannot provide Python 2.
  - INFO: Executable: Not found

## Python 3 provider (optional)
  - INFO: pyenv: Path: /opt/pyenv/libexec/pyenv
  - INFO: pyenv: Root: /opt/pyenv
  - INFO: Using: g:python3_host_prog = "/home/jakobgm/.virtualenvs/NeoVim3/bin/python"
  - INFO: Executable: /home/jakobgm/.virtualenvs/NeoVim3/bin/python
  - INFO: Python version: 3.6.6
  - INFO: pynvim version: 0.3.2
  - OK: Latest pynvim is installed.

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.6.1p33 (2019-01-30 revision 66950) [x86_64-linux]
  - INFO: Host: /home/jakobgm/.gem/ruby/2.5.0/bin/neovim-ruby-host
  - OK: Latest "neovim" gem is installed: 0.8.0

## Node.js provider (optional)
  - INFO: Node.js: v10.6.0
  - INFO: Neovim node.js host: /home/jakobgm/.node_modules/lib/node_modules/neovim/bin/cli.js
```

**Comment by jalvesaq** [comment]
I can't replicate the issue. Do you have other markdown plugins installed?

**Comment by JakobGM** [comment]
Hi, @jalvesaq, thanks for the extremely quick reply.

I'm sorry to have wasted your time. You are completely correct, it was another markdown plugin. Specifically [this one](https://github.com/vim-pandoc/vim-rmarkdown)

```
Plug 'vim-pandoc/vim-rmarkdown' " RMarkdown support
```

Thanks again!

**Comment by nhejazi** [comment]
I've somehow run into this error recently -- was the solution to just disable the `vim-rmarkdown` plugin?

**Comment by jalvesaq** [comment]
If you have vim-rmarkdown installed, you could try to uninstall it. If the issue is fixed after that, you have to choose between Nvim-R and vim-rmarkdown (or make a pull request to fix the incompatibility).

**Comment by nhejazi** [comment]
Thank you for clarifying so quickly @jalvesaq! I ran into this issue from the linked issue (#394), which mentioned trying to uninstall vim-rmarkdown and then re-install, but this didn't work for me. I've since removed vim-rmarkdown (I use Nvim-R far more centrally in my workflow) and the issue appears resolved. I would try to fix it but am not familiar enough with either of these packages to be able to do so effectively.
